### PR TITLE
code for check

### PR DESCRIPTION
**Pull request by kristinagerdt**

### CODE REVIEW

File: src/GenericArray.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+public class GenericArray<T> implements Iterable<T> {
+
+    private T[] elementsArr;
+    private int size;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Class<T> type;
+
+    GenericArray(Class<T> cs) {
+        @SuppressWarnings("unchecked") final T[] array = (T[]) Array.newInstance(cs, DEFAULT_CAPACITY);
```
Comment (MichelRogov) [inline src/GenericArray.java]: I think you could use new Object[] and cast it to T[] i.e. all return statements or in constructor. Actually if you would check original class ArrayList, it does the same.

File: src/GenericArray.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    private void resizeCapacity(int minCapacity) {
+        int oldCapacity = elementsArr.length;
+        int newCapacity = oldCapacity + (oldCapacity * 2);
+        if (newCapacity - minCapacity < 0)
+            newCapacity = minCapacity;
+        elementsArr = copyArray(elementsArr, newCapacity);
+    }
+
+    private T[] copyArray(T[] elementsArr, int newSize) {
+        @SuppressWarnings("unchecked") final T[] newArr = (T[]) Array.newInstance(type, newSize);
```
Comment (MichelRogov) [inline src/GenericArray.java]: same as above

File: src/GenericArray.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    private Class<T> type;
+
+    GenericArray(Class<T> cs) {
+        @SuppressWarnings("unchecked") final T[] array = (T[]) Array.newInstance(cs, DEFAULT_CAPACITY);
+        this.elementsArr = array;
+        this.type = cs;
+    }
+
+    public GenericArray(Class<T> cs, int capacity) {
+        @SuppressWarnings("unchecked") final T[] array = (T[]) Array.newInstance(cs, capacity);
```
Comment (MichelRogov) [inline src/GenericArray.java]: same as above

File: src/GenericArray.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    public T removeByIndex(int index) {
+        if (!rangeCheck(index)) {
+            return null;
+        }
+        T oldValue = elementsArr[index];
+        copyArray(elementsArr, index + 1, elementsArr, index, size - index);
+        size--;
+        fillNull();
```
Comment (MichelRogov) [inline src/GenericArray.java]: Are you sure, that you need to iterate the whole array just by removing one element? Maybe elementsArr[--size]=null; would be better here?

File: src/GenericArray.java (hunk 1 above)
Comment (MichelRogov) [reply src/GenericArray.java]: Yep, much better

**Comment by MichelRogov** [comment]
Didn't you forget trimToSize function?

**Comment by kristinagerdt** [comment]
trimToSize() на месте после функции fillNull()

**Comment by MichelRogov** [comment]
Ok, didn’t see it

**Pull request merged**